ibverbs/src/lib.rs

```diff
@@ -464,8 +445,20 @@ impl Context {
     }
 
     /// Returns the valid GID table entries of this RDMA device context.
-    pub fn gid_table(&self) -> &[GidEntry] {
-        &self.gid_table
+    pub fn gid_table(&self) -> io::Result<Vec<GidEntry>> {
+        let mut gid_table = vec![ffi::ibv_gid_entry::default(); MAX_GID_TBL_LEN];
+        let num_entries = unsafe {
```

Comment:
This needs to check for whether `num_entries < 0` and if so error still I think.